File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Resources/resourcecost.rb
module Asposetasksjava
  module ResourceCost
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'project.mpp')

		resources = project.getResources().toList()

		rsc = Rjb::import('com.aspose.tasks.Rsc')

		# Parse through all the collected tasks
		i = 0
		while i < resources.size()
			resource = resources.get(i)
		    if resource.get(rsc.NAME) != nil
				puts "Cost: " + resource.get(rsc.COST).to_string
			    puts "ACWP: " + resource.get(rsc.ACWP).to_string
			    puts "BCWS: " + resource.get(rsc.BCWS).to_string
			    puts "BCWP: " + resource.get(rsc.BCWP).to_string
			    puts "---------------------------------------------"
			end
		    i +=1
		end
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithResources/ResourceCost.php
<?php
namespace Aspose\Tasks\WorkingWithResources;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Rsc as Rsc;
class ResourceCost{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

		$resources = $project->getResources()->toList();

		$rsc = new Rsc();

		# Parse through all the collected tasks
		$i = 0;
		while ($i < sizeof($resources)) {
            $resource = $resources -> get($i);
		    if ($resource -> get($rsc -> NAME) != null) {
                print "Cost: " . (string)$resource -> get($rsc -> COST).PHP_EOL; //. to_string
			    print "ACWP: " . (string)$resource -> get($rsc -> ACWP).PHP_EOL;//. to_string
			    print "BCWS: " . (string)$resource -> get($rsc -> BCWS).PHP_EOL; //. to_string
			    print "BCWP: " . (string)$resource -> get($rsc -> BCWP).PHP_EOL;// . to_string
			    print "---------------------------------------------".PHP_EOL;
			}
		    $i += 1;
		}

    }

}
?>